SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -219,7 +219,10 @@ private static void MergeSettingsFromCommandLine(AnalysisConfig config, Processe
 
             foreach (Property item in args.GetAllProperties())
             {
-                config.SetExplicitValue(item.Id, item.Value); // this will overwrite the setting if it already exists
+                if (!ProcessRunnerArguments.ContainsSensitiveData(item.Id) && !ProcessRunnerArguments.ContainsSensitiveData(item.Value))
```

Comment:
The pre-processor doesn't make any assumptions about what is considered sensitive data - it leaves that entirely up to the ProcessRunnerArguments. It checks both id and value to make sure nothing it is writing to the file is considered sensitive data.